_extras/design.md

```diff
@@ -4,35 +4,159 @@ title: Lesson Design
+- argue for or against strategies to to normalize data
+- formulate techniques beyond transformations to clean data (exploit correlations, create summary statistics, ...)
+
+### Design and train a Deep Neural Network
+
+This includes knowledge of when to different types of layers
+
+After this module, learners can ...
+
+- list/repeat the three ingrediants to a feed forward network: input, hidden layers, output
+- classify parts of a feed forward network
+- describe a fully connected (dense) layer
+- describe a convolutional layer
+- describe a max pooling layer
+- describe an activation function
+- describe a softmax layer
+- argue against abundant use of the sigmoid function (exploding/vanishing gradients)
+- select the output data shape from a convolutional layer
+- interpret errors with convolutional layers
+- execute a 3 layer network on the MNIST data (or similar)
+- differentiate a dense layer and a convolutional layer
+- experiment with values of dense layer and a convolutional layer
+- select a layer type depending on the input data
+- develop a 5 layer network that comprises both layer types
+
+### Troubleshoot the learning process
+
+Often when designing neural networks training will not automatically work very well. This requires setting the parameters of the training algorithm correctly, modifying the design of the network or changing the data pre-processing
+
+### Measure the performance of the network
+
+After training the performance of the network should be checked to prevent overfitting.
+
+After this module, learners can ...
+
+- define precision and recall/accuracy for a classification task
+- state that cross-validation is used in Deep Learning too
+- describe how to split a dataset into training/test/validation set
+- describe how Drop-Out Layers work
+- execute a plot to draw the loss per epoch for training and test set
+- compare values of precision and recall
+- differentiate a overfitting network from a well-behaved network
+- detect a underfitting network
+- critique a provided network design
+
+### Visualizing Data and Results
+
+Within each episode how to visualize data and results
+
+After this module, learners can ...
+
+- identify important plots to create at the end of training (provide selected samples and their prediction)
+- execute plotting of important variables during training (loss, ROC)
+- use tensorboard and related callbacks during training 
+- examine the results of a partners network
+- critique the results of a partners network
+
+### Re-use existing network architectures with and without pre-trained weights
+
+Re-use of architectures is common in deep learning. Especially when using pre-trained weights (transfer-learning) it can also be very powerful.
+
+After this module, learners can ...
+
+- describe what transfer learning stands for
+- explain the expected relation of data sets (one used for the network originally and one for the network at hand)
```

Comment:
maybe word it slightly different:
explain in what situations transfer learning is beneficial.

Reply:
fixed with d5fe467